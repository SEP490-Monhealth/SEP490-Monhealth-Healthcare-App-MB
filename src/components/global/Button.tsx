import React from "react"

import { ActivityIndicator, Text, TouchableOpacity, View } from "react-native"

import { cn } from "@/lib/utils"

interface ButtonProps {
  onPress?: () => void
  children: React.ReactNode
  iconStart?: React.ReactNode
  iconEnd?: React.ReactNode
  variant?: "primary" | "secondary" | "ghost" | "danger"
  size?: "sm" | "md" | "lg"
  loading?: boolean
  disabled?: boolean
  icon?: boolean
  className?: string
}

export const Button: React.FC<ButtonProps> = ({
  onPress,
  children,
  iconStart,
  iconEnd,
  variant = "primary",
  size = "md",
  loading = false,
  disabled = false,
  icon = false,
  className
}) => {
  const variants = {
    primary: disabled ? "bg-gray-400" : "bg-primary",
    secondary: disabled ? "bg-gray-300" : "bg-gray-700",
    ghost: disabled ? "bg-gray-300" : "bg-[#eaeaea]",
    danger: disabled ? "bg-gray-300" : "bg-red-600"
  }

  const sizes = {
    sm: "px-4 py-2 text-sm",
    md: "px-5 py-3 text-base",
    lg: "px-6 py-4 text-lg"
  }

  const variantClass = variants[variant]
  const sizeClass = sizes[size]

  return (
    <TouchableOpacity
      className={cn(
        "flex flex-row items-center justify-center rounded-[10px]",
        variantClass,
        sizeClass,
        className,
        icon ? "p-3" : ""
      )}
      onPress={onPress}
      disabled={disabled || loading}
      activeOpacity={0.8}
    >
      {loading ? (
        <ActivityIndicator color="#fff" />
      ) : icon ? (
        <View className="flex items-center justify-center">{children}</View>
      ) : (
        <View className="flex flex-row items-center">
          {iconStart && <View className="mr-2">{iconStart}</View>}
          {children && (
            <Text className={cn("font-semibold text-white")}>{children}</Text>
          )}
          {iconEnd && <View className="ml-2">{iconEnd}</View>}
        </View>
      )}
    </TouchableOpacity>
  )
}
