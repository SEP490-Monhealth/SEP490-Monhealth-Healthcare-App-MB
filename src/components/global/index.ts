export { Button } from "./Button"
export { VStack, HStack } from "./Stack"
