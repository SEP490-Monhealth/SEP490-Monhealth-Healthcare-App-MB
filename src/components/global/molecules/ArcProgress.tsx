import React from "react"

import { Text, View } from "react-native"
import { AnimatedCircularProgress } from "react-native-circular-progress"

import { COLORS } from "@/constants/appConstants"

import { cn } from "@/lib/utils"

interface CenterCircleProps {
  size: number
  innerSizeRatio: number
  calorieValue: number
  maxCalories: number
  label: string
}

const CenterCircle = ({
  size,
  innerSizeRatio,
  calorieValue,
  maxCalories,
  label
}: CenterCircleProps) => {
  return (
    <View
      className="absolute left-1/2 top-1/2 -translate-x-1/2 -translate-y-1/2 transform select-none items-center justify-center rounded-full bg-white"
      style={{
        width: size * innerSizeRatio,
        height: size * innerSizeRatio,
        borderWidth: 14,
        borderColor: "#fcfcfc"
      }}
    >
      <Text className="font-tbold text-2xl text-typography">
        {calorieValue} / {maxCalories}
      </Text>
      <Text className="font-tmedium text-base text-accent">{label}</Text>
    </View>
  )
}

interface ArcProgressProps {
  size: number
  width: number
  fill?: number
  tintColor?: string
  backgroundColor?: string
  arcSweepAngle: number
  rotation: number
  centerCircle?: boolean
  calorieValue: number
  maxCalories: number
  label: string
  className?: string
}

export const ArcProgress = ({
  size,
  width,
  fill = 0,
  tintColor = COLORS.primary,
  backgroundColor = "#e2e8f0",
  arcSweepAngle = 360,
  rotation = 0,
  centerCircle = false,
  calorieValue = 0,
  maxCalories = 0,
  label = "",
  className = ""
}: ArcProgressProps) => {
  return (
    <View className={cn("relative items-center", className)}>
      <AnimatedCircularProgress
        size={size}
        width={width}
        fill={fill}
        prefill={fill}
        tintColor={tintColor}
        backgroundColor={backgroundColor}
        arcSweepAngle={arcSweepAngle}
        rotation={rotation}
        lineCap="round"
      />

      {centerCircle && (
        <CenterCircle
          size={size}
          innerSizeRatio={0.75}
          calorieValue={calorieValue}
          maxCalories={maxCalories}
          label={label}
        />
      )}
    </View>
  )
}
