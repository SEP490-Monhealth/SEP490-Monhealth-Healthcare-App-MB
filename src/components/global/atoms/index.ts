import { Avatar } from "./Avatar"
import { Button } from "./Button"
import { Calendar } from "./Calendar"
import { Card } from "./Card"
import { Checkbox } from "./Checkbox"
import { Container } from "./Container"
import { Content } from "./Content"
import { Input } from "./Input"
import { Progress } from "./Progress"
import { Schedule } from "./Schedule"
import { ScrollArea } from "./ScrollArea"
import { Select } from "./Select"
import { HStack, VStack } from "./Stack"
import { Tabs, TabsContent, TabsList, TabsTrigger } from "./Tabs"

export {
  Avatar,
  Button,
  Calendar,
  Card,
  Checkbox,
  Container,
  Content,
  Input,
  Progress,
  Schedule,
  ScrollArea,
  Select,
  VStack,
  HStack,
  Tabs,
  TabsContent,
  TabsList,
  TabsTrigger
}
