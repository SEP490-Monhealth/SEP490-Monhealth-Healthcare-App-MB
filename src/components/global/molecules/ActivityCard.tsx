import React from "react"

import { Image, Text, TouchableOpacity } from "react-native"

import { toFixed } from "@/utils/formatters"

import { Card, CardHeader, Checkbox, HStack, VStack } from "../atoms"

interface ActivityCardProps {
  name: string
  durationMinutes: number
  caloriesBurned: number
  isCompleted: boolean
  onPress?: () => void
  onCheckboxChange?: (value: boolean) => void
}

export const ActivityCard = ({
  name,
  durationMinutes,
  caloriesBurned,
  isCompleted,
  onPress,
  onCheckboxChange
}: ActivityCardProps) => {
  const handlePress = () => {
    if (onCheckboxChange) {
      if (onCheckboxChange) {
        onCheckboxChange(!isCompleted)
      }
    }
  }

  return (
    <Card onPress={handlePress}>
      <HStack className="w-full items-center justify-between">
        <HStack center gap={12} onPress={onPress}>
          <TouchableOpacity
            activeOpacity={1}
            className="flex h-12 w-12 items-center justify-center rounded-full bg-muted"
          >
            <Image
              source={require("../../../../public/icons/activities/dumbbell.png")}
              style={{ width: 24, height: 24 }}
            />
          </TouchableOpacity>

          <VStack gap={0}>
            <HStack center gap={8}>
              <CardHeader label={name} />
            </HStack>

            <Text className="font-tmedium text-sm text-accent">
              {toFixed(caloriesBurned)} kcal • {durationMinutes ?? 0} phút
            </Text>
          </VStack>
        </HStack>

        <Checkbox checked={isCompleted} onCheckChange={onCheckboxChange} />
      </HStack>
    </Card>
  )
}
