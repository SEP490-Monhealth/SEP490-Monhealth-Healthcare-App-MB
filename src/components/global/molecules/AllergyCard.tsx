import React from "react"

import { Text } from "react-native"

import { MoreHorizontal } from "lucide-react-native"

import { COLORS } from "@/constants/color"

import { Card, HStack } from "../atoms"
import { IconButton } from "./IconButton"

interface AllergyCardProps {
  variant?: "default" | "more"
  name: string
  isSelected?: boolean
  onPress?: () => void
  onMorePress?: () => void
}

export const AllergyCard = ({
  variant = "default",
  name,
  isSelected,
  onPress,
  onMorePress
}: AllergyCardProps) => {
  return (
    <Card
      className={`rounded-2xl border-2 ${
        isSelected ? "border-primary" : "border-border"
      }`}
      onPress={onPress}
    >
      <HStack className="w-full items-center justify-between">
        <Text className="font-tmedium text-lg text-primary">{name}</Text>

        {variant === "more" && (
          <IconButton
            size="sm"
            icon={<MoreHorizontal size={20} color={COLORS.primary} />}
            onPress={onMorePress}
          />
        )}
      </HStack>
    </Card>
  )
}
