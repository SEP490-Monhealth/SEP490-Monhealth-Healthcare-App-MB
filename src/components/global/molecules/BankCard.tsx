import React from "react"

import { Text, TouchableOpacity, View } from "react-native"
import { SvgUri } from "react-native-svg"

import {
  ArchiveAdd,
  ArchiveMinus,
  ArchiveTick,
  FilterAdd,
  FilterTick
} from "iconsax-react-native"
import { Bold } from "lucide-react-native"

import { COLORS } from "@/constants/color"

import { useSelectBankStore } from "@/stores/bankStore"

import { Card, VStack } from "../atoms"
import { IconButton } from "./IconButton"

interface BankCardProps {
  code: string
  name: string
  shortName: string
  logoUrl: string
  onPress?: () => void
}

export const BankCard = ({
  code,
  name,
  shortName,
  logoUrl,
  onPress
}: BankCardProps) => {
  const selectedCode = useSelectBankStore((state) => state.code)

  const isSelected = selectedCode === code
  return (
    <Card className={`${isSelected && "border border-primary"}`}>
      <VStack gap={12}>
        <View className="flex-1 flex-row items-center justify-between gap-2">
          <View className="flex-1 flex-row gap-2">
            <TouchableOpacity
              activeOpacity={1}
              className="h-12 w-12 items-center justify-center rounded-full bg-muted"
            >
              <SvgUri uri={logoUrl} width={24} height={24} />
            </TouchableOpacity>

            <View className="flex-1 flex-col">
              <Text className="font-tmedium text-base text-primary">
                {shortName}
              </Text>

              <Text
                className="mt-1 font-tmedium text-sm text-accent"
                numberOfLines={1}
                ellipsizeMode="tail"
              >
                {name}
              </Text>
            </View>
          </View>

          <IconButton
            size="sm"
            icon={
              isSelected ? (
                <ArchiveTick variant="Bold" size={20} color={COLORS.primary} />
              ) : (
                <ArchiveAdd variant="Linear" size={20} color={COLORS.primary} />
              )
            }
            onPress={onPress}
          />
        </View>
      </VStack>
    </Card>
  )
}
