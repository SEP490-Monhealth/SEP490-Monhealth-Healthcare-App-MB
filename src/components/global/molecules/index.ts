import { ActivityCard } from "./ActivityCard"
import { ArcProgress } from "./ArcProgress"
import { BankCard } from "./BankCard"
import { BookingCard } from "./BookingCard"
import { BookingItem } from "./BookingItem"
import { CertificateCard } from "./CertificateCard"
import { ChatCard } from "./ChatCard"
import { CircularProgress } from "./CircularProgress"
import { ConsultantBankCard } from "./ConsultantBankCard"
import { ConsultantCard } from "./ConsultantCard"
import { CounterText } from "./CounterText"
import { CustomHeader } from "./CustomHeader"
import { ErrorDisplay } from "./ErrorDisplay"
import { ExerciseCard } from "./ExerciseCard"
import { FileCard } from "./FileCard"
import { FoodCard } from "./FoodCard"
import { IconButton } from "./IconButton"
import { ListFooter } from "./ListFooter"
import { ListHeader } from "./ListHeader"
import { MealCard } from "./MealCard"
import { MeetingCard } from "./MeetingCard"
import { MetricCard } from "./MetricCard"
import { NotificationCard } from "./NotificationCard"
import { NutritionCard } from "./NutritionCard"
import { NutritionItem } from "./NutritionItem"
import { RatingStars } from "./RatingStars"
import { ReviewCard } from "./ReviewCard"
import { ScheduleCard } from "./ScheduleCard"
import { ScheduleExceptionCard } from "./ScheduleExceptionCard"
import { ServiceChip } from "./ServiceChip"
import { StepHeader } from "./StepHeader"
import { SubscriptionCard } from "./SubscriptionCard"
import { TimeSlotSelector } from "./TimeSlotSelector"
import { WaterReminderCard } from "./WaterReminderCard"
import { WeightCard } from "./WeightCard"
import { WithdrawalRequestCard } from "./WithdrawalRequestCard"
import { WorkoutCard } from "./WorkoutCard"

export {
  ActivityCard,
  ArcProgress,
  BankCard,
  BookingCard,
  BookingItem,
  CertificateCard,
  ChatCard,
  CircularProgress,
  ConsultantBankCard,
  ConsultantCard,
  CounterText,
  CustomHeader,
  ErrorDisplay,
  ExerciseCard,
  FileCard,
  FoodCard,
  IconButton,
  ListFooter,
  ListHeader,
  MealCard,
  MeetingCard,
  MetricCard,
  NotificationCard,
  NutritionCard,
  NutritionItem,
  RatingStars,
  ReviewCard,
  ScheduleCard,
  ScheduleExceptionCard,
  ServiceChip,
  StepHeader,
  SubscriptionCard,
  TimeSlotSelector,
  WaterReminderCard,
  WeightCard,
  WithdrawalRequestCard,
  WorkoutCard
}
