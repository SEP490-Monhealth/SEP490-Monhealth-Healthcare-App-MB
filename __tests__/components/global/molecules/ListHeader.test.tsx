import React from "react"

import { View } from "react-native"

import { render, screen } from "@testing-library/react-native"

import { ListHeader } from "@/components/global/molecules"

// Mock cn utility
jest.mock("@/lib/utils", () => ({
  cn: jest.fn((...classes: string[]) => classes.join(" "))
}))

describe("ListHeader Component", () => {
  it("renders children correctly", () => {
    render(
      <ListHeader>
        <View testID="child-view" />
      </ListHeader>
    )

    expect(screen.getByTestId("child-view")).toBeTruthy()
  })

  it("applies default class name", () => {
    render(<ListHeader />)

    expect(screen.getByTestId("root-view").props.className).toContain(
      "bg-background"
    )
  })

  it("appends additional className prop", () => {
    render(<ListHeader className="custom-class" />)

    expect(screen.getByTestId("root-view").props.className).toContain(
      "bg-background custom-class"
    )
  })
})
