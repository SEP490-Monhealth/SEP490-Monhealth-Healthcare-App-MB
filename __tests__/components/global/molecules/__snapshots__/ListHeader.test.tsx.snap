// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeader applies default class names 1`] = `
<View
  className="bg-background"
>
  <View
    testID="header-view"
  />
</View>
`;

exports[`ListHeader merges custom className with default className 1`] = `
<View
  className="bg-background custom-class"
>
  <View
    testID="header-view"
  />
</View>
`;
